packages/react/src/primitives/shared/styleUtils.ts

```diff
@@ -44,31 +49,52 @@ export const convertGridSpan = (
   }
   // ResponsiveObject<PropertyType>
   if (typeof spanValue === 'object' && spanValue != null) {
-    const newObj: ResponsiveObject<string> = {};
-    Object.entries(spanValue).forEach(([key, value]) => {
-      newObj[key] = getGridSpan(value as GridSpanType);
-    });
-    return newObj;
+    return (Object.entries(spanValue) as Array<[string, GridSpanType]>).reduce(
+      (acc, [key, value]) => {
+        return {
+          ...acc,
+          [key]: getGridSpan(value),
+        };
+      },
+      {} as ResponsiveObject<string>
+    );
```

Comment:
```suggestion
    return (Object.entries(spanValue) as Array<[string, GridSpanType]>).reduce(
      (acc, [key, value]) => ({ ...acc, [key]: getGridSpan(value) }),
      {} as ResponsiveObject<string>
    );
```

Reply:
Jeez my reduce code isn't good enough for you

Reply:
I'm just trying to reduce it 🥁